plugins/package-managers/maven/src/main/kotlin/Tycho.kt

```diff
@@ -0,0 +1,237 @@
+import org.apache.maven.execution.MavenSession
+import org.apache.maven.project.MavenProject
+
+import org.codehaus.plexus.DefaultPlexusContainer
+import org.codehaus.plexus.PlexusContainer
+import org.codehaus.plexus.classworlds.ClassWorld
+import org.codehaus.plexus.logging.BaseLoggerManager
+
+import org.eclipse.aether.graph.DependencyNode
+
+import org.ossreviewtoolkit.analyzer.AbstractPackageManagerFactory
+import org.ossreviewtoolkit.analyzer.PackageManager
+import org.ossreviewtoolkit.analyzer.PackageManagerResult
+import org.ossreviewtoolkit.analyzer.ProjectResults
+import org.ossreviewtoolkit.model.DependencyGraph
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.model.config.RepositoryConfiguration
+import org.ossreviewtoolkit.model.utils.DependencyGraphBuilder
+import org.ossreviewtoolkit.plugins.packagemanagers.maven.utils.DefinitionFileResolveFunc
+import org.ossreviewtoolkit.plugins.packagemanagers.maven.utils.DependencyTreeLogger
+import org.ossreviewtoolkit.plugins.packagemanagers.maven.utils.LocalProjectWorkspaceReader
+import org.ossreviewtoolkit.plugins.packagemanagers.maven.utils.MavenDependencyHandler
+import org.ossreviewtoolkit.plugins.packagemanagers.maven.utils.MavenLogger
+import org.ossreviewtoolkit.plugins.packagemanagers.maven.utils.MavenSupport
+import org.ossreviewtoolkit.plugins.packagemanagers.maven.utils.identifier
+import org.ossreviewtoolkit.plugins.packagemanagers.maven.utils.internalId
+import org.ossreviewtoolkit.plugins.packagemanagers.maven.utils.isTychoProject
+import org.ossreviewtoolkit.plugins.packagemanagers.maven.utils.parseMavenOutput
+import org.ossreviewtoolkit.plugins.packagemanagers.maven.utils.toOrtProject
+import org.ossreviewtoolkit.utils.ort.createOrtTempFile
+
+/**
+ * A package manager implementation supporting Maven projects using [Tycho](https://github.com/eclipse-tycho/tycho).
+ */
+class Tycho(
+    name: String,
+    analysisRoot: File,
+    analyzerConfig: AnalyzerConfiguration,
+    repoConfig: RepositoryConfiguration
+) : PackageManager(name, "Tycho", analysisRoot, analyzerConfig, repoConfig) {
+    class Factory : AbstractPackageManagerFactory<Tycho>("Tycho") {
+        override val globsForDefinitionFiles = listOf("pom.xml")
+
+        override fun create(
+            analysisRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) = Tycho(type, analysisRoot, analyzerConfig, repoConfig)
+    }
+
+    /**
+     * The builder to generate the dependency graph. This could actually be a local variable, but it is also needed
+     * to construct the final [PackageManagerResult].
+     */
+    private lateinit var graphBuilder: DependencyGraphBuilder<DependencyNode>
+
+    override fun mapDefinitionFiles(definitionFiles: List<File>): List<File> = definitionFiles.filter(::isTychoProject)
```

Comment:
If this is sufficient here, why is `definitionFiles.filterNot(::isTychoProject)` not sufficient in the `Maven` class?

Reply:
In contrast to Maven, Tycho only processes the root pom of a multi-module build. Only for those, `isTychoProject()` returns *true*. For Maven, the child poms under the Tycho project roots need to be filtered out explicitly.